src/SharedMoudule/Components/Filter/Filter.tsx

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import styles from "./Filter.module.css";
+import { baseUrl } from "../../../Constants/Components/Urls";
+import { useEffect, useState } from "react";
+
+export default function Filter({ setUsersList }) {
+  const [searchByName, setSearchByName] = useState("");
```

Comment:
I suggest using searchParams for handling states related to searching and filtering and pagination.
You can look for `useSearchParams` in react-router-dom.

This will make both the user and developer experience far better, as the state will be stored in the URL and you can refresh the page to get the current state without resetting every thing from the begining.